nbs/10_compute_correlations/05_gtex_v8/py/50-combine_all.py

```diff
@@ -0,0 +1,246 @@
+
+# %% [markdown] tags=[]
+# # Modules loading
+
+# %% tags=[]
+import numpy as np
+import pandas as pd
+
+from clustermatch import conf
+
+# %% [markdown] tags=[]
+# # Settings
+
+# %% tags=["parameters"]
+DATASET_CONFIG = conf.GTEX
+# whole blood by default, but this is a parameters cells that can be changed when running papermill
+GTEX_TISSUE = "whole_blood"
+GENE_SEL_STRATEGY = "var_pc_log2"
+
+# %% tags=[]
+assert GTEX_TISSUE is not None, "Tissue not selected"
+
+# %% [markdown] tags=[]
+# # Paths
+
+# %% tags=[]
+INPUT_GENE_EXPR_DATA_FILE = (
+    DATASET_CONFIG["GENE_SELECTION_DIR"]
+    / f"gtex_v8_data_{GTEX_TISSUE}-{GENE_SEL_STRATEGY}.pkl"
+)
+display(INPUT_GENE_EXPR_DATA_FILE)
+
+assert INPUT_GENE_EXPR_DATA_FILE.exists()
+
+# %% tags=[]
+INPUT_CORR_FILE_TEMPLATE = (
+    DATASET_CONFIG["SIMILARITY_MATRICES_DIR"]
+    / DATASET_CONFIG["SIMILARITY_MATRIX_FILENAME_TEMPLATE"]
+)
+display(INPUT_CORR_FILE_TEMPLATE)
+
+# %% tags=[]
+OUTPUT_FILE = DATASET_CONFIG["SIMILARITY_MATRICES_DIR"] / str(
+    INPUT_CORR_FILE_TEMPLATE
+).format(
+    tissue=GTEX_TISSUE,
+    gene_sel_strategy=GENE_SEL_STRATEGY,
+    corr_method="all",
+)
+display(OUTPUT_FILE)
+
+# %% [markdown] tags=[]
+# # Load data
+
+# %% [markdown] tags=[]
+# ## Gene Ensembl ID -> Symbol mapping
+
+# %% tags=[]
+gene_map = pd.read_pickle(
+    DATASET_CONFIG["DATA_DIR"] / "gtex_gene_id_symbol_mappings.pkl"
+)
+
+# %% tags=[]
+gene_map = gene_map.set_index("gene_ens_id")["gene_symbol"].to_dict()
+
+# %% tags=[]
+assert gene_map["ENSG00000145309.5"] == "CABS1"
+
+# %% [markdown] tags=[]
+# ## Gene expression
+
+# %% tags=[]
+data = pd.read_pickle(INPUT_GENE_EXPR_DATA_FILE)
+
+# %% tags=[]
+data.shape
+
+# %% tags=[]
+data.head()
+
+# %% [markdown] tags=[]
+# ## Clustermatch
+
+# %% tags=[]
+clustermatch_df = pd.read_pickle(
+    str(INPUT_CORR_FILE_TEMPLATE).format(
+        tissue=GTEX_TISSUE,
+        gene_sel_strategy=GENE_SEL_STRATEGY,
+        corr_method="clustermatch",
+    )
+)
+
+# %% tags=[]
+clustermatch_df.shape
+
+# %% tags=[]
+clustermatch_df.head()
+
+# %% tags=[]
+assert data.index.equals(clustermatch_df.index)
+
+# %% [markdown] tags=[]
+# ## Pearson
+
+# %% tags=[]
+pearson_df = pd.read_pickle(
+    str(INPUT_CORR_FILE_TEMPLATE).format(
+        tissue=GTEX_TISSUE,
+        gene_sel_strategy=GENE_SEL_STRATEGY,
+        corr_method="pearson",
+    )
+)
+
+# %% tags=[]
+pearson_df.shape
+
+# %% tags=[]
+pearson_df.head()
+
+# %% tags=[]
+assert data.index.equals(pearson_df.index)
+
+# %% [markdown] tags=[]
+# ## Spearman
+
+# %% tags=[]
+spearman_df = pd.read_pickle(
+    str(INPUT_CORR_FILE_TEMPLATE).format(
+        tissue=GTEX_TISSUE,
+        gene_sel_strategy=GENE_SEL_STRATEGY,
+        corr_method="spearman",
+    )
+)
+
+# %% tags=[]
+spearman_df.shape
+
+# %% tags=[]
+spearman_df.head()
+
+# %% tags=[]
+assert data.index.equals(spearman_df.index)
+
+
+# %% [markdown] tags=[]
+# ## Merge
+
+# %% tags=[]
+def get_upper_triag(data, k=1):
```

Comment:
Calling this parameter `data` might be a little bit confusing, since you have a dataframe above called `data` for the gene expression data which you don't seem to be using this function for.

Maybe you could call it `similarity_data` or something like that to avoid the ambiguity?

Reply:
Great point, I will change it. Actually, this function is useful in other cases, so I'll move into the library.